### PR TITLE
risc-v/boot: verbosely log boot errors

### PR DESCRIPTION
**Pull request by axel-h**
Log an explicit error message for everything that can fail during boot. This is helpful for debugging, e.g. when porting the kernel to a new platform or changing the system configuration.

### CODE REVIEW

**Comment by axel-h** [comment]
The preprocess test fail because of the `printf` call that this patch adds.

**Review by lsf37 — APPROVED** [review]
These look good from my side.

Should not affect the proofs since it's all boot code, but let me start a testboard just to make sure.

**Comment by lsf37** [comment]
@ssrg-bamboo `proof`

**Comment by nomadeel** [comment]
@ssrg-bamboo `proof`

**Comment by ssrg-bamboo** [comment]
Hello, I'm a bot! I've set up a proof testboard for this PR [here](https://github.com/seL4/gh-testboard/blob/Hensoldt-Cyber-patch-axel-20-429-050711/manifest.xml), and results should show up as `pending` in status checks in a few minutes.

**Comment by nomadeel** [comment]
I guess the bot didn't receive the first message as a notification.

**Comment by axel-h** [comment]
Can I merge this now or is there another manual step needed to update the preprocess templates?

**Comment by mbrcknl** [comment]
> Can I merge this now or is there another manual step needed to update the preprocess templates?

A preprocess test failure just compares this pull request with the current head to determine whether this PR potentially impacts the proofs. Once you merge this, it's the new head, so it will be the basis for future preprocess tests.

TL;DR: no need to do anything to update the preprocess check. We just need a reasonable level of confidence this won't break the proofs. And I'm reasonably confident of that.

Not sure what's going on with the simulation tests, though. I guess most are timeouts? Except maybe clang armv6a, which seems unrelated to this PR.

**Review by mbrcknl — APPROVED** [review]

**Comment by lsf37** [comment]
Let me re-run the simulations. Only `armv6a` seems to have actually failed, the others were cancelled because of that failure. There is one test in the test set that is dodgy on GH (see #seL4/ci-actions#112).

**Comment by lsf37** [comment]
To add on to Matt's explanation: it's Ok to merge it, i.e. no other prework or preconditions, but there is an extra step afterwards (which we can do).

Basically, the preprocess-base is the last known-good version that has a proof, and that only bumps automatically to other preprocess-equal versions or if we explicitly tell it to.

That latter part is the "preprocess bump" script that people might have mentioned before. It lives in https://github.com/seL4/l4v/tree/master/misc/bump together with a bit more explanation.

**Comment by lsf37** [comment]
Ok, all simulation runs successful now. It does look like that failure is not that rare and we should do something about it.

**Comment by lsf37** [comment]
(preprocess bump done as well)
